python/ray/train/torch/train_loop_utils.py

```diff
@@ -467,7 +467,15 @@ def get_device(self) -> torch.device:
         if torch.cuda.is_available():
             gpu_ids = ray.get_gpu_ids()
             if len(gpu_ids) > 0:
-                device_id = gpu_ids[0]
+                # Some backwards compatibility
+                try:
+                    local_rank = session.get_local_rank()
+                except Exception:
+                    local_rank = train.local_rank()
+                # https://github.com/pytorch/pytorch/blob/35563f4fcd28e486cc5
+                # 8053acc15fe280123e7be/torch/distributed/launch.py#L72-L97
+                device_id = local_rank
+                logger.debug(f"setting device id {device_id} as local rank.")
```

Comment:
@amogkam Unless I set this to local_rank, the benchmark script will error out.

If you print `get_gpu_ids()`, you get (2, 3) (len=2). However, you still need to set the device to local_rank or else you get a runtime error cuda invalid device ordinal.

I took a look at the blame which made this change here: https://github.com/ray-project/ray/commit/029517a037b1219423ab45af79db1e9296bc39c7

which is obviously different in logic but not quite sure what you were trying to do in the previous logic.

Reply:
Yeah this is a bug…will be fixed by https://github.com/ray-project/ray/pull/26493

using local_rank directly won’t work for fractional or multiple gpus per worker